File: packages/contracts/src/organization-team.ts
import type { VersionedRecord } from "./common";
import type { OrganizationMembershipRoleCode } from "./organization-membership";
import type { UserStatus } from "./user";

export interface OrganizationTeamMemberRecord {
  user_id: string;
  display_name: string;
  email: string;
  status: UserStatus;
  role_code: OrganizationMembershipRoleCode;
  role_label: string;
}

export interface OrganizationTeamRecord extends VersionedRecord {
  organization_id: string;
  name: string;
  description: string | null;
  member_count: number;
  members: OrganizationTeamMemberRecord[];
}

export interface OrganizationTeamUpsertRequest {
  name: string;
  description?: string | null;
  member_user_ids: string[];
}

File: apps/api/app/schemas/organization_team.py
from uuid import UUID

from pydantic import BaseModel, ConfigDict

from app.db.models.user import UserStatus
from app.schemas.common import BaseReadModel


class OrganizationTeamMemberRead(BaseModel):
    user_id: UUID
    display_name: str
    email: str
    status: UserStatus
    role_code: str
    role_label: str


class OrganizationTeamRead(BaseReadModel):
    organization_id: UUID
    name: str
    description: str | None
    member_count: int
    members: list[OrganizationTeamMemberRead]

    model_config = ConfigDict(from_attributes=True)


class OrganizationTeamUpsertRequest(BaseModel):
    name: str
    description: str | None = None
    member_user_ids: list[UUID]
